Ical.Net/Evaluation/RecurrencePatternEvaluator.cs

```diff
@@ -162,10 +162,20 @@ private IEnumerable<CalDateTime> EnumerateDates(CalDateTime originalDate, CalDat
         var dateCount = 0;
         while (true)
         {
-            if (searchEndDate < GetIntervalLowerLimit(intervalRefTime, pattern))
+            var lowerLimit = GetIntervalLowerLimit(intervalRefTime, pattern, originalDate);
+
+            if (searchEndDate < lowerLimit)
                 break;
 
-            var candidates = GetCandidates(intervalRefTime, pattern, expandBehavior);
+            // Use the original DTSTART month as the anchor for YEARLY rules when BYMONTH is not specified.
+            // This ensures BYMONTHDAY expansion/limiting is performed relative to the DTSTART month
+            // (the expected behavior for yearly rules without an explicit BYMONTH), instead of the
+            // current interval reference month.
+            var anchorMonth = pattern is { Frequency: FrequencyType.Yearly, ByMonth.Count: 0 }
+                ? originalDate.Month
```

Comment:
When changing this to return `intervalRefTime.Month` in both cases, all tests still run through. Would be great to have a test case where this distinction is essential.

Reply:
The anchorMonth was one of the ideas introduced in a very first draft. It was only partly helpful. When investigating further, anchorMonth became fully redundant - which I oversaw. Excellent finding, Please keep up these reviews.